test/core/base/cache.js

```diff
@@ -1,32 +1,180 @@
-describe('axe._cache', function() {
+describe('axe._cache', function () {
   'use strict';
 
-  it('should set items from the cache without error', function() {
-    function fn() {
-      axe._cache.set('foo', 'bar');
-    }
-    assert.doesNotThrow(fn);
-  });
+  describe('set', function () {
+    it('should set items without error', function () {
+      function fn() {
+        axe._cache.set('foo', 'bar');
+      }
+      assert.doesNotThrow(fn);
+    });
+
+    // undefined is a valid result type of a potentially expensive operation in axe-core
```

Comment:
You all have convinced me that you believe this is true. You have yet to provide a valid use case for caching nothing though. Why is this necessary?

Reply:
Both @WilcoFiers and I have agreed that we want this functionality

Reply:
Why? I don't understand why an explanation is too much to ask.

Reply:
My understanding is that the two of you spoke offline on this question. I'm resolving this thread so we can finalize review.

Reply:
This has still not been resolved. We did speak on Slack, but the question posted at https://github.com/dequelabs/axe-core/pull/3608#discussion_r965041066 has never been answered.